File: palang-virtual-machine/src/assembly/assembly.rs
use std::collections::HashMap;

use super::{function::Function, model::Model, prompt::Prompt};

#[derive(Debug, Clone)]
pub struct Assembly {
    pub name: String,
    pub models: HashMap<String, Model>,
    pub prompts: HashMap<String, Prompt>,
    pub functions: HashMap<String, Function>
}

impl Assembly {
    pub fn new() -> Self {
        Assembly {
            name: String::new(),
            models: HashMap::new(),
            prompts: HashMap::new(),
            functions: HashMap::new(),
        }
    }
}

File: vs-code-debugger/src/models/assembly.ts
import { Model } from "./model";
import { Prompt } from "./prompt";
import { Function } from "./function";

export class Assembly {
    constructor(
        public name: string,
        public sourceFile: string,
        public models: Array<Model>,
        public prompts: Array<Prompt>,
        public functions: Array<Function>,
    ) {}
}
